rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/db/sql/DBSQLFile.kt

```diff
@@ -31,29 +32,45 @@ class DBSQLFile(private val name: String, private val connection: Connection) :
     }
 
     override fun readEqual(key: Value): Record {
-        if (resultSet == null) {
-            chain(emptyList())
+        return readEqual(toFields(key))
+    }
+
+    private fun filterRecord(result: Record): Record {
+        return if (result.matches(lastKey)) {
+            result
+        } else {
+            resultSet?.last()
```

Comment:
Is this acceptable to have a null resultSet here?

Reply:
At the moment filterRecord is called just when resultSet is not null, but, as I think it could be used elsewhere, I think we can have a "relaxed" behaviour. The idea is "in case you have an open resultset, just set on EOF, if you don't have it, EOF is already on". Do you think it wolud be better to encapsulate this in a fun with a speaking name?

Reply:
I am not sure as I do not know the context very well, but if you would be "surprised" if resultSet is actually null then I would write `resultSet!!.last()` and if later you confirm it can be relaxed then I would change it back